kb-dk/ds-image
# Disc 292 fullsize image

Thumbnail limit is width>150 or size > 150 (see ds-image-behaviour.yaml)

ds-image will log each image call with result of "thumbnail" parsing, so it is easy to test (enable debug logging)
DEBUG: Image presentation type was parsed as thumbnail=true from parameters for FIF=/DAMJP2/DAM/Samlingsbilleder/0000/750/978/DNF_1955-00298_00092

This implementation is very conservative and will conclude thumbnail also if some of the non size-parameters are defined. 
It is better to be conservative and later loosen up than giving too much control over thumbnail extraction.



IIIF:
thumbnail :http://devel11.statsbiblioteket.dk:10001/ds-image/v1/IIIF/%2FDAMJP2%2FDAM%2FSamlingsbilleder%2F0000%2F750%2F978%2FDNF_1955-00298_00092/full/%21150%2C150/0/default.jpg
full size : http://devel11.statsbiblioteket.dk:10001/ds-image/v1/IIIF/%2FDAMJP2%2FDAM%2FSamlingsbilleder%2F0000%2F750%2F978%2FDNF_1955-00298_00092/full/max/0/default.jpg


IIP:
thumbnail: http://devel11:10001/ds-image/v1/IIP/?FIF=/DAMJP2/DAM/Samlingsbilleder/0000/750/978/DNF_1955-00298_00092&HEI=150&WID=150&CVT=jpeg
full size: http://devel11:10001/ds-image/v1/IIP/?FIF=/DAMJP2/DAM/Samlingsbilleder/0000/750/978/DNF_1955-00298_00092&CVT=jpeg

Any zoom requests:
Will always be fullsize


Licensemodule:
The licensemodule has license giving  access to: search, fullsize and thumbnails (http://devel11:10001/ds-license/license.jsp?licenseId=1679482985150)
It makes sense that if you have access to fullsize you also have to thumbnails. And this is controlled by licensemodule and not ds-image where it would be hardcoded.
You can disable this license (expire data: 2020)

The licensemodule also have a license only access is search, thumbnail  (http://devel11:10001/ds-license/license.jsp?licenseId=1705408647844)

Easy way to test:
1)ds-search: http://devel11:10001/ds-search/
2) search for: id:"ds.samlingsbilleder:f30cd1d0-f53b-11e9-aa5a-00505688346e"
Thumbnail will be shown. Click title to see full size
3) Disable fullsize license: /devel11:10001/ds-license/license.jsp?licenseId=1679482985150
4) new private window
5) search again. Thumbnail will be shown
6) Full size will not work
